libraries/Microsoft.Bot.Builder/BotState.cs

```diff
@@ -215,7 +215,7 @@ private class CachedBotState
         {
             public CachedBotState(IDictionary<string, object> state = null)
             {
-                State = state ?? new ConcurrentDictionary<string, object>();
```

Comment:
>ConcurrentDictionary [](start = 37, length = 20)

The description doesn't describe why this is the correct thing to do.  Can someone please fill in more details...?